DSL/conditions.md

```diff
@@ -0,0 +1,288 @@
+ {"path": ["type"], "operation": "contains", "value": "any"}
+]
+```
+
+- First condition: Checks if the root of the schema has a key `type`.
+- Second condition: Checks if the array `type` contains the value "any".
+
+### 10. Check that an object property exists, that is an array, and it contains an item of a given type
+
+```json
+{    
+    "type": [ {"const": "hello"}, "number", "string" ]
+}
+```
+```json
+[
+ {"path": ["type"], "operation": "type-equals", "value": "array"},
+ {"path": ["type"], "operation": "contains-type", "value": "object"}
+]
+```
+
+- First condition: Checks if the value of `type` is of type array.
+- Second condition: Checks if the array `type` contains any elements of type object.
+
+### 11. Check that an object property exists, that it is an object, and that any of its values have a specific property
+
+```json
+{
+  "dependencies": {
+    "foo": { "type": "object" }
+  }
+}
+```
+```json
+[
+ {"path": ["dependencies", {"pattern": ".*"}], "operation": "has-key", "value": "type"}
+]
+```
+
+- Condition: Checks if any value within the `dependencies` object has a key `type`.
+
+### 12. Check that an object property exists, that it is an object, and that any of its values are of a specific type
+
+```json
+{
+  "dependencies": {
+    "foo": { "type": "object" },
+    "bar": [ "foo" ]
+  }
+}
+```
+```json
+[
+ {"path": ["dependencies", {"pattern": ".*"}], "operation": "type-is", "value": "array"}
+]
+```
+
+- Condition: Checks if any value within the `dependencies` object is of type array.
+
+### 13. Check that an object property exists and its size
+
+```json
+{    
+    "$ref": "#/$defs/foo",
+    "$defs":{ "foo": {} }
+}
+```
+```json
+[
+ {"path": [], "operation": "has-key", "value": "$ref"},
+ {"path": [], "operation": "size-equals", "value": 1}
+]
+```
+
+- First condition: Checks if the root of the schema has a key `$ref`.
+- Second condition: Checks if the root schema has exactly 1 property.
+
+### 14. Check that an object property exists, that it is an object, and that a value is an array of a certain size
+
+```json
+{    
+    "prefixItems": []
+}
+```
+```json
+[
+ {"path": [], "operation": "has-key", "value": "prefixItems"},
+ {"path": ["prefixItems"], "operation": "size-equals", "value": 0}
+]
+```
+
+- First condition: Checks if the root of the schema has a key `prefixItems`.
+- Second condition: Checks if the array `prefixItems` has exactly 0 items.
+
+### 15. Check that an object property is a real number that represents an integer
+
+```json
+[
+ {"path": ["properties", "baz"], "operation": "type-equals", "value": "integer"}
+]
+```
+
+- Condition: Checks if the value of properties.baz is an integer.
+
+### 16.  Check that an object property exists, that it is an object, and that any of its keys represent Perl regular expressions that are not also ECMA
+
+### 17.  Check that an object property exists and that is represents a Perl regular expression that is not ECMA
+
+16 and 17 are to be implemented
```

Comment:
if something is not implemented yet, don't mention it in the docs. You can create a GitHub Issue instead and add it here when its actually implemented